test/mainnet-fork/Uniswapv4.t.sol

```diff
@@ -864,7 +864,109 @@ contract MainnetControllerUniswapV4Tests is ForkTestBase {
+
+        IncreasePositionResult memory mintResult     = _mintPosition(tickLower, tickUpper, liquidity, type(uint160).max, type(uint160).max);
+        DecreasePositionResult memory decreaseResult = _decreasePosition(mintResult.tokenId, mintResult.liquidityIncrease, 0, 0);
+
+        uint256 valueDeposited = _to18From6Decimals(mintResult.amount0Spent)        + _to18From6Decimals(mintResult.amount1Spent);
+        uint256 valueReceived  = _to18From6Decimals(decreaseResult.amount0Received) + _to18From6Decimals(decreaseResult.amount1Received);
+
+        // Value difference should be less than 1e18 which including protocol fees and rounding errors
+        // 2 wei at 6 decimals is 2e12 at 18 decimals
+        assertApproxEqAbs(valueReceived, valueDeposited, 2e12);
+    }
+
+    function testFuzz_increaseAndDecreaseFullAmounts(
+        int24   tickLower,
+        int24   tickUpper,
+        uint128 initialLiquidity
+    ) external {
+        tickLower = int24(_bound(int256(tickLower), -10_000, 10_000 - 1));
+
+        int256 boundedUpperMax = int256(tickLower) + 1_000 > 10_000 ? int256(10_000) : int256(tickLower) + 1_000;
+
+        tickUpper = int24(_bound(int256(tickUpper), int256(tickLower) + 1, boundedUpperMax));
+        initialLiquidity = uint128(_bound(uint256(initialLiquidity), 1e6, 2_000_000e6));
+
+        uint128 additionalLiquidity = initialLiquidity / 2;
+
+        vm.startPrank(SPARK_PROXY);
+        mainnetController.setUniswapV4TickLimits(_POOL_ID, -10_000, 10_000, 1_000);
+        rateLimits.setRateLimitData(_DEPOSIT_LIMIT_KEY,  2_000_000e18, uint256(2_000_000e18) / 1 days);
+        rateLimits.setRateLimitData(_WITHDRAW_LIMIT_KEY, 2_000_000e18, uint256(2_000_000e18) / 1 days);
+        vm.stopPrank();
+
+        IncreasePositionResult memory mintResult     = _mintPosition(tickLower, tickUpper, initialLiquidity, type(uint160).max, type(uint160).max);
+        IncreasePositionResult memory increaseResult = _increasePosition(mintResult.tokenId, additionalLiquidity, type(uint160).max, type(uint160).max);
+
+        uint256 valueBeforeIncrease = _to18From6Decimals(mintResult.amount0Spent)     + _to18From6Decimals(mintResult.amount1Spent);
+        uint256 valueAdded          = _to18From6Decimals(increaseResult.amount0Spent) + _to18From6Decimals(increaseResult.amount1Spent);
+        uint256 totalValueDeposited = valueBeforeIncrease + valueAdded;
+
+        uint128 totalLiquidity = mintResult.liquidityIncrease + increaseResult.liquidityIncrease;
+
+        DecreasePositionResult memory decreaseResult = _decreasePosition(mintResult.tokenId, totalLiquidity, 0, 0);
+
+        uint256 valueReceived  = _to18From6Decimals(decreaseResult.amount0Received) + _to18From6Decimals(decreaseResult.amount1Received);
+
+        // Value difference should be less than 1e12 which including protocol fees and rounding errors
+        // 10 wei at 6 decimals is 1e13 at 18 decimals
+        assertApproxEqAbs(totalValueDeposited, valueReceived, 1e13);
```

Comment:
Try 2 instead of 1e13 here and see if it works

Reply:
tried that it goes to 7wei the diff